### PR TITLE
Add Tekton manifests

### PR DESCRIPTION
**Pull request by JohnNiang**
### What this PR dose / Why we need it

Add Tekton manifests, including
- ServiceAccount for PipelineRun
- Shared storage
- Pipeline
- Trigger Template
- Trigger

Some common manifests will be hosted at [ks-infra](https://github.com/kubesphere-sigs/ks-infra), please see upstream PR: https://github.com/kubesphere-sigs/ks-infra/pull/6.

BTW, I also disabled GitHub workflow for pull request.

/kind feature
/cc @kubesphere-sigs/sig-devops

### CODE REVIEW

File: .github/tekton/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# KS Repo CI/CD
+
+Why dose `ks-releaser` project have a folder `.github/tekton`? Cuz we want to replace GitHub workflow with Tekton.
```
Comment (LinuxSuRen) [inline .github/tekton/README.md:3]: `Cuz` ?

File: .github/tekton/pull-request-pipeline.yaml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        - name: revision
+          value: $(params.revision)
+      workspaces:
+        - name: output
+          workspace: repo
+    - name: build
+      runAfter:
+        - checkout
+      taskRef:
+        name: goreleaser-and-trivy
```
Comment (LinuxSuRen) [inline .github/tekton/pull-request-pipeline.yaml:48]: We already put the common tasks into ks-infra. It's necessary to let all contributors know that.

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]

File: .github/tekton/README.md (hunk 1 above)
Comment (JohnNiang) [reply .github/tekton/README.md:3]: image

Comment (LinuxSuRen) [reply .github/tekton/README.md:3]: As you can see, this is an informal word. Please use a formal word instead.

**Review by LinuxSuRen — COMMENTED** [review]
/lgtm
/approve

File: .github/tekton/pull-request-pipeline.yaml (hunk 2 above)
Comment (JohnNiang) [reply .github/tekton/pull-request-pipeline.yaml:48]: I've added a document about those common task, please refer to https://github.com/kubesphere-sigs/ks-infra/tree/master/prod/ks-devops-ext-tekton-common#common-tasks.

**Pull request merged**